ui/src/design-system/components/button/button.tsx

```diff
@@ -39,15 +41,15 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
         className={classNames(styles.button, {
           [styles.success]: theme === ButtonTheme.Success,
           [styles.plain]: theme === ButtonTheme.Plain,
-          [styles.disabled]: disabled,
+          [styles.disabled]: disabled ?? loading,
         })}
-        disabled={disabled}
+        disabled={disabled ?? loading}
```

Comment:
Not really part of this PR, but recently read a twitter thread, that I can't find again, where they discussed a lot about the accessibility of disabled buttons..
Found this one when doing a quick googling now https://axesslab.com/disabled-buttons-suck/

Reply:
Yes this is an interesting topic, had a nerdy chat with my UX colleagues in Umeå. I agree most of the time it's better to avoid disabled buttons. In our case, we use disabled state on the deployment save button (if we have no changes to save yet or if data is missing or malformed). One challenge for us is the form has multiple sections, but the save button is shared. For example, how can we communicate there is an issue in a section that is currently not in view?

Something to think about for sure, thanks for bringing it up!